password_enhancements.module

```diff
@@ -228,6 +229,52 @@ function _password_enhancements_get_roles_from_form_state(array $form, FormState
   return array_merge([AccountInterface::AUTHENTICATED_ROLE], $roles, (!empty($input['roles']) ? array_values($input['roles']) : []));
 }
 
+/**
+ * Returns a list of constraints for a set of roles.
+ *
+ * @param string[] $roles
+ *   Array of role IDs.
+ *
+ * @return \Drupal\password_enhancements\PasswordConstraintInterface[]
+ *   Array of password constraints.
+ */
+function _password_enhancements_load_constraints_by_roles(array $roles): array {
```

Comment:
Can't we make this somehow an class method instead? I am not saying yet that this should be a service, I am saying this should not be a procedural function. It could be even a static method, not sure yet in which class, I am in the beginning of my review.

Reply:
I was also fiddling with this idea – but which class should we add this to?
- This is not exactly an AccessControlHandler, so it doesn't belong there.
- This is not an Annotation, an EventSubscriber, an Exception, a Form, a Logger nor a Plugin, so it doesn't belong to any of them.
- PasswordConstraintBase is a base for all plugins, and this is not about any single plugin.
- PasswordConstraintManager is about managing all the plugins available on the site, and this is not about all the plugins.
- PasswordPolicy is bound to one role, so it doesn't belong there.
- PasswordChecker seems to be the only utility-like class, but as it's also a service, I'm a bit hesitant to pollute it with this as a static method.

Reply:
How about introducing a new `PasswordConstraint` utility service for this purpose? Initially, I wanted to suggest a simple class with static methods - so not a service - but because this method depends on the EntityTypeManager service thanks for the `PasswordPolicy::loadMultipleByRoles($roles)` call we should create a service. If we do that then in tests we can mock what `loadMultipleByRoles()` returns (or throws) to test edge-cases.